### ISSUE TITLE
Failed to convert .pb to .tflite

### ISSUE THREAD

**Issue by ttfleet** [tensorflow/tensorflow#23772]
When I convert a pb file to tflite, I meet a error as follow.  what's wrong?

(tensorflow) XXXX :~/tensorflow/tensorflow-master$ bazel-bin/tensorflow/contrib/lite/toco/toco  \
>   --input_file=./headpose.pb \
>   --input_format=TENSORFLOW_GRAPHDEF  --output_format=TFLITE \
>   --output_file=./headpose.tflite --inference_type=FLOAT \
>   --input_type=FLOAT --input_arrays=input \
>   --output_arrays=output –input_shapes=1,416,416,3
2018-11-15 21:19:52.503499: W tensorflow/contrib/lite/toco/toco_cmdline_flags.cc:282] --input_type is deprecated. It was an ambiguous flag that set both --input_data_types and --inference_input_type. If you are trying to complement the input file with information about the type of input arrays, use --input_data_type. If you are trying to control the quantization/dequantization of real-numbers input arrays in the output file, use --inference_input_type.
2018-11-15 21:19:52.616030: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: RefSwitch
2018-11-15 21:19:52.633837: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: AssignSub
2018-11-15 21:19:52.633907: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: RefSwitch
2018-11-15 21:19:52.633920: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: AssignSub
2018-11-15 21:19:52.634233: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: RefSwitch
2018-11-15 21:19:52.634249: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: AssignSub
2018-11-15 21:19:52.634269: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: RefSwitch
2018-11-15 21:19:52.634280: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: AssignSub
2018-11-15 21:19:52.634410: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: RefSwitch
2018-11-15 21:19:52.634424: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: AssignSub
2018-11-15 21:19:52.634444: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: RefSwitch
2018-11-15 21:19:52.634454: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: AssignSub
2018-11-15 21:19:52.634767: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: RefSwitch
2018-11-15 21:19:52.634782: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: AssignSub
2018-11-15 21:19:52.634801: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: RefSwitch
2018-11-15 21:19:52.634811: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: AssignSub
2018-11-15 21:19:52.635636: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: RefSwitch
2018-11-15 21:19:52.635656: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: AssignSub
2018-11-15 21:19:52.635686: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: RefSwitch
2018-11-15 21:19:52.635705: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: AssignSub
2018-11-15 21:19:52.639048: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: RefSwitch
2018-11-15 21:19:52.639092: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: AssignSub
2018-11-15 21:19:52.639121: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: RefSwitch
2018-11-15 21:19:52.639131: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: AssignSub
2018-11-15 21:19:52.652115: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: RefSwitch
2018-11-15 21:19:52.652212: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: AssignSub
2018-11-15 21:19:52.652237: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: RefSwitch
2018-11-15 21:19:52.652246: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: AssignSub
2018-11-15 21:19:52.682724: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: RefSwitch
2018-11-15 21:19:52.682769: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: AssignSub
2018-11-15 21:19:52.682804: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: RefSwitch
2018-11-15 21:19:52.682826: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1110] Converting unsupported operation: AssignSub
2018-11-15 21:19:52.691380: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before Removing unused ops: 437 operators, 737 arrays (0 quantized)
2018-11-15 21:19:52.699329: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] After Removing unused ops pass 1: 293 operators, 513 arrays (0 quantized)
2018-11-15 21:19:52.706308: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before general graph transformations: 293 operators, 513 arrays (0 quantized)
2018-11-15 21:19:52.706484: F tensorflow/contrib/lite/toco/graph_transformations/resolve_tensorflow_switch.cc:98] Check failed: other_op->type == OperatorType::kMerge Found BatchNormalization as non-selected output from Switch, but only Merge supported.
Aborted (core dumped)

**Comment by ymodak** [comment]
Can you try with latest tf-nightly build? Also can you please provide **System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
- TensorFlow version:
- Python version:
- Installed using virtualenv? pip? conda?:
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**Comment by ttfleet** [comment]
OS Platform and Distribution i5+1050ti+win10+anaconda:
TensorFlow installed from (binary):
TensorFlow version:1.12.0

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ymodak: It has been 17 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by ymodak** [comment]
Is this still an issue for you? Did you try latest tf-nightly build?
You can try setting virtual env and test if haven't already:
> virtualenv -p python3 venv-tf-nightly     #creates env named tf-nightly
> source venv-tf-nightly/bin/activate   #sources the newly created env
> pip3 install tf-nightly    #here you build required versions

**Comment by ymodak** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by jiangxiluning** [comment]
HI, @ymodak I used the nightly build， same error occurred.

**Comment by ymodak** [comment]
@jiangxiluning Can you please post a new issue describing your problem? Also make sure you provide all the information asked by the template. Thanks!

**Comment by WilliamL1** [comment]
> the
I have almost the same problem with @jiangxiluning, tflite does not support switching and merging, can you please give some inspiration? Here are the details of my issue. [LINK](https://github.com/tensorflow/tensorflow/issues/27241)
